pkg/discovery/dtofactory/node_entity_dto_builder.go

```diff
@@ -444,10 +446,27 @@ func (builder *nodeEntityDTOBuilder) getAffinityCommoditiesSold(node *api.Node,
 		commoditiesSold = append(commoditiesSold, commodityDTO)
 	}
 
-	for _, workloadKey := range hostnameSpreadWorkloads.UnsortedList() {
+	var podQualifiedNames sets.String
+	if len(hostnameSpreadWorkloads) > 0 {
+		// Concatenate runningPods and pendingPods into a single slice and extract pod name
+		allPods := append(builder.clusterSummary.GetPendingPodsOnNode(node), builder.clusterSummary.GetRunningPodsOnNode(node)...)
+		for _, pod := range allPods {
+			podQualifiedNames.Insert(fmt.Sprintf("%s/%s", pod.Namespace, pod.Name))
+		}
+	}
+
+	var used float64
+	for workloadKey := range hostnameSpreadWorkloads {
+		workloadKeyPods := hostnameSpreadWorkloads[workloadKey]
```

Comment:
for workloadKey,workloadKeyPods:= range hostnameSpreadWorkloads {

Reply:
Fixed. Thanks!